db/redis.go

```diff
@@ -969,3 +986,170 @@ func (r *RedisDriver) InsertMitre(years []string) error {
 
 	return nil
 }
+
+// GetAdvisoriesPaloalto get AdvisoryID: []CVE IDs
+func (r *RedisDriver) GetAdvisoriesPaloalto() (map[string][]string, error) {
+	return r.getAdvisories(fmt.Sprintf(advKeyFormat, models.PaloaltoType))
+}
+
+// CountPaloalto count paloalto table
+func (r *RedisDriver) CountPaloalto() (int, error) {
+	depstr, err := r.conn.HGet(context.Background(), depKey, models.PaloaltoType).Result()
+	if err != nil {
+		if errors.Is(err, redis.Nil) {
+			return 0, nil
+		}
+		return 0, err
+	}
+
+	// deps: {"AdvisoryID": {"CVEID": {"part#vendor#product": {}}}}
+	var deps map[string]map[string]map[string]struct{}
+	if err := json.Unmarshal([]byte(depstr), &deps); err != nil {
+		return 0, xerrors.Errorf("Failed to unmarshal JSON. err: %w", err)
+	}
+
+	count := 0
+	for _, cs := range deps {
+		count += len(cs)
+	}
+
+	return count, nil
+}
+
+// InsertPaloalto Cve information from DB.
+func (r *RedisDriver) InsertPaloalto(advs iter.Seq2[models.Paloalto, error]) error {
+	ctx := context.Background()
+
+	batchSize := viper.GetInt("batch-size")
+	if batchSize < 1 {
+		return fmt.Errorf("Failed to set batch-size. err: batch-size option is not set properly")
+	}
+	var err error
+
+	advtoCVEs := map[string][]string{}
```

Comment:
This information seems to be included in newDeps, if so, can we remove this and use newDeps instead?